src/app/common/hooks/use-background-location-redirect.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { useLocationState } from '@app/common/hooks/use-location-state';
+
+/*
+when modals are opened in a new tab they lose the location.state.backgroundLocation
+ this hook sets the backgroundLocation to be RouteUrls.Home to improve UX
+*/
+export function useBackgroundLocationRedirect() {
+  const { pathname, state } = useLocation();
+  const navigate = useNavigate();
+  const backgroundLocation = useLocationState('backgroundLocation');
+
+  useEffect(() => {
+    void (async () => {
+      switch (true) {
+        // FIXME ReceiveCollectibleOrdinal loses state?.btcAddressTaproot in a new tab
+        // this can be improved to try and fetch btcAddressTaproot
+        case pathname === RouteUrls.ReceiveCollectibleOrdinal && !state?.btcAddressTaproot:
```

Comment:
For `RouteUrls.ReceiveCollectibleOrdinal`, it relies on `btcAddressTaproot` that is set in `location.state`. 

If you open a new tab this state is lost and although the modal opens, the `btcAddressTaproot` is lost so it's unusable. 

For now I just send users back to `RouteUrls.Home`. This could probably be improved by refactoring how we get `btcAddressTaproot` in `ReceiveCollectibleOrdinal`

Reply:
Not a big fan of using a `switch(true)`. Why not just use `if`s? 

This is better for readability imo. My thoughts map to the first comment of the HN post of a guy who wrote an article doing this (https://news.ycombinator.com/item?id=26777090) re favouring early return.

Separately—while identical behaviour of modals when doing a fresh load of the app would be ideal—if we need to add a lot of logic to handle this, then I wouldn't consider it a dealbreaker if the under layer doesn't load correctly. Contextually, in case of sign out, it isn't so important.

Reply:
I like using `switch(true)` as I think its cleaner and more elegant than a lot of `if` statements.  I can change it though and use early returns here and in future to match the code base.

If we are OK with having an empty screen behind all modals opened in a new tab then we can avoid the addition of this logic.

It will look like this, with home tabs but no tab content: 

![Screenshot 2023-07-31 at 11 16 23](https://github.com/hirosystems/wallet/assets/2938440/2a74a6fc-c406-4eb6-8f98-df976d92408b)

For `RouteUrls.ReceiveCollectibleOrdinal` we will need to do something though as we lose track of the taproot address when opening in a new tab and the modal shows but is non-functional as there is no address to copy:
![Screenshot 2023-07-31 at 11 21 10](https://github.com/hirosystems/wallet/assets/2938440/4aae1ea1-b8de-4e1c-9378-d46dca5dc9b1)